docs/source/usage_guides/deepspeed_multiple_model.md

```diff
@@ -0,0 +1,215 @@
+```python
+with deepspeed.zero.Init(accelerator.deepspeed_plugin.config):
+    model = MyModel(...)
+```
+
+### Training
+
+From here, your training loop can be whatever you like, making sure that `teacher_model` is never being trained on:
+
+```python
+teacher_model.eval()
+student_model.train()
+for batch in train_dataloader:
+    with torch.no_grad():
+        output_teacher = teacher_model(**batch)
+    output_student = student_model(**batch)
+    # Combine the losses or modify it in some way
+    loss = output_teacher.loss + output_student.loss
+    accelerator.backward(loss)
+    optimizer.step()
+    scheduler.step()
+    optimizer.zero_grad()
+```
+
+## Multiple models all being trained
+
+Multiple models being trained is a more complicated scenario and is actively being looked upon by the team.
+In its current state, we assume that each model **completely disjoint** from the other during training.
+
+With this situation, we still require two [`utils.DeepSpeedPlugin`]'s to be made, however we also further
+require a second `Accelerator`, since we need to call the proper `deepspeed` engine, and `Accelerator` only
+carries one instance of it at a time.
+
+Since the [`state.AcceleratorState`] is a stateful object though, it is already aware of both `DeepSpeedPlugin`'s available,
+meaning you can just instantiate a second `Accelerator` with no extra arguments:
+
+```python
+accelerator_0 = Accelerator(deepspeed_plugin=deepspeed_plugins)
+accelerator_1 = Accelerator()
+```
+
+Similar to before, we can call either `accelerator_0.state.enable_deepspeed_plugin()` to enable/disable
+a particular plugin, and call `.prepare` like normal:
+
+```python
+# can be `accelerator_0`, `accelerator_1`, or by calling `AcceleratorState().enable_deepspeed_plugin(...)`
+accelerator_0.state.enable_deepspeed_plugin("model1")
+model_0 = AutoModel.from_pretrained(...)
+# For this example, `get_training_items` is a nonexistent function that gets the setup we need for training
+optimizer_0, scheduler_0, train_dl, eval_dl = get_training_items(model_0)
+model_0, optimizer_0, scheduler_0, train_dl, eval_dl = accelerator.prepare(model_0, optimizer_0, scheduler_0, train_dl, eval_dl)
+
+accelerator_1.state.enable_deepspeed_plugin("model2")
+model_1 = AutoModel.from_pretrained(...)
+# For this example, `get_training_items` is a nonexistent function that gets the setup we need for training
+optimizer_1, scheduler_1, _, _ = get_training_items(model1)
+model_1, optimizer_1, scheduler_1 = accelerator.prepare(model1, optimizer1, scheduler1)
+```
+
+And now you can train:
+
+```python
+for batch in dl:
+    outputs_1 = model_0(**batch)
+    accelerator_0.backward(outputs_1.loss)
+    optimizer_0.step()
+    scheduler_0.step()
+    optimizer_0.zero_grad()
+    outputs_2 = model_1(**batch)
+    accelerator_1.backward(outputs_2.loss)
+    optimizer_1.step()
+    scheduler_1.step()
+    optimizer_1.zero_grad()
+```
```

Comment:
I find this example very difficult to follow since the numbers don't match.

perhaps once you explained why this is so and they have to be that way, non-numerical differentiator would work better? perhaps the same teacher/student naming would be more appropriate?

also we are facing humans here, model_0 and model_1 isn't how we think - we think model_1 and model_2 instead. I'd suggest to leave 0-indexing to computer indices.

Reply:
also adding some vertical white space to group things together would help the readability in this last code section.

Reply:
Went with `first_` and `second_` to be more readable (and the prior example used words)